00_Base/src/datasources/CdrsDatasource.ts

```diff
@@ -0,0 +1,29 @@
+import { Inject, Service, Token } from 'typedi';
+import { PaginatedResult } from '../model/PaginatedResult';
+import { ITransactionDatasource } from './ITransactionDatasource';
+import { TRANSACTION_DATASOURCE_SERVICE_TOKEN } from './TransactionDatasource';
+import { ICdrsDatasource } from './ICdrsDatasource';
+import { CdrMapper } from '../mapper/cdr.mapper';
+import { Cdr } from '../model/Cdr';
+
+export const CDR_DATASOURCE_SERVICE_TOKEN = new Token('CDR_DATASOURCE_SERVICE_TOKEN');
+
+@Service(CDR_DATASOURCE_SERVICE_TOKEN)
+export class CdrsDatasource implements ICdrsDatasource {
+  constructor(
+    @Inject(TRANSACTION_DATASOURCE_SERVICE_TOKEN)
+    private readonly transactionDatasource: ITransactionDatasource,
+    private readonly cdrMapper: CdrMapper
+  ) {
+  }
+
+  async getCdrs(cpoCountryCode: string, cpoPartyId: string, mspCountryCode: string, mspPartyId: string, dateFrom?: Date, dateTo?: Date, offset?: number, limit?: number): Promise<PaginatedResult<Cdr>> {
```

Comment:
Your call, but I kind of like the return type that I've seen @yuvenus use where we return something like `[Cdr[], int]`.

Then, on the other side, it would look like: `const [cdrs, total] = datasource.getCdrs(...)`.

But what you have rn is nice cause of how explicit it is so idk.